api/src/main/java/org/projectnessie/tools/catalog/migration/api/CatalogMigrator.java

```diff
@@ -0,0 +1,247 @@
+      throw new UnsupportedOperationException(
+          String.format(
+              "target catalog %s doesn't implement SupportsNamespaces to create missing namespaces.",
+              targetCatalog().name()));
+    }
+
+    if (!(sourceCatalog() instanceof SupportsNamespaces)) {
+      throw new UnsupportedOperationException(
+          String.format(
+              "source catalog %s doesn't implement SupportsNamespaces to list all namespaces.",
+              sourceCatalog().name()));
+    }
+
+    if (deleteEntriesFromSourceCatalog() && sourceCatalog() instanceof HadoopCatalog) {
+      throw new UnsupportedOperationException(
+          "Source catalog is a Hadoop catalog and it doesn't support deleting the table entries just from the catalog. Please configure `deleteEntriesFromSourceCatalog` as `false`");
+    }
+  }
+
+  private static final Logger LOG = LoggerFactory.getLogger(CatalogMigrator.class);
+  private final ImmutableCatalogMigrationResult.Builder resultBuilder =
+      ImmutableCatalogMigrationResult.builder();
+  private final Set<Namespace> processedNamespaces = new HashSet<>();
+
+  /**
+   * Get the table identifiers which matches the regular expression pattern input from all the
+   * namespaces.
+   *
+   * @param identifierRegex regular expression pattern. If null, fetches all the table identifiers
+   *     from all the namespaces.
+   * @return Set of table identifiers.
+   */
+  public Set<TableIdentifier> getMatchingTableIdentifiers(String identifierRegex) {
+    LOG.info("Collecting all the namespaces from source catalog...");
+    Set<Namespace> namespaces = new LinkedHashSet<>();
+    getAllNamespacesFromSourceCatalog(Namespace.empty(), namespaces);
+
+    Predicate<TableIdentifier> matchedIdentifiersPredicate;
+    if (identifierRegex == null) {
+      LOG.info("Collecting all the tables from all the namespaces of source catalog...");
+      matchedIdentifiersPredicate = tableIdentifier -> true;
+    } else {
+      LOG.info(
+          "Collecting all the tables from all the namespaces of source catalog"
+              + " which matches the regex pattern:{}",
+          identifierRegex);
+      Pattern pattern = Pattern.compile(identifierRegex);
+      matchedIdentifiersPredicate =
+          tableIdentifier -> pattern.matcher(tableIdentifier.toString()).matches();
+    }
+    return namespaces.stream()
+        .flatMap(
+            namespace -> {
+              try {
+                return sourceCatalog().listTables(namespace).stream()
+                    .filter(matchedIdentifiersPredicate);
+              } catch (IllegalArgumentException | NoSuchNamespaceException exception) {
+                if (namespace.isEmpty()) {
+                  // some catalogs don't support default namespace.
+                  // Hence, just log the warning and ignore the exception.
+                  LOG.warn(
+                      "Failed to identify tables from default namespace: {}",
```

Comment:
`default` is wrong here, same for the comment

Reply:
changed to `empty` namespace